src/main/java/org/opensearch/security/user/UserService.java

```diff
@@ -97,6 +98,7 @@ private static CType getUserConfigName() {
     public UserService(ClusterService clusterService, ConfigurationRepository configurationRepository, Settings settings, Client client) {
         this.clusterService = clusterService;
         this.configurationRepository = configurationRepository;
+        this.passwordHasher = new BCryptPasswordHasher();
```

Comment:
I think we may want to have a central place where this is instantiated once and passed to components that need it. Can we look into instantiating this within [OpenSearchSecurityPlugin.createComponents](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java#L1017-L1030)?

Reply:
That makes sense, thank you!